src/openfl/display/SubBitmapData.hx

```diff
@@ -0,0 +1,310 @@
+	private var __texX1 = 1.0;
+	private var __texY0 = 0.0;
+	private var __texY1 = 1.0;
+	private var __texX:Int;
+	private var __texY:Int;
+	private var __texWidth:Int;
+	private var __texHeight:Int;
+	private var __parentBitmap:BitmapData;
+	private var __rotated = false;
+
+	public function new(atlasBitmap:BitmapData, x:Int, y:Int, width:Int, height:Int, frameX:Int, frameY:Int, frameWidth:Int, frameHeight:Int, rotated:Bool) {
+		#if !display
+		var w = width, h = height;
+		var offsetX = 0;
+		var offsetY = 0;
+		if (frameWidth > 0 && frameHeight > 0) {
+			offsetX = -frameX;
+			offsetY = -frameY;
+			w = frameWidth;
+			h = frameHeight;
+		} else if (rotated) {
+			var tmp = w;
+			w = h;
+			h = tmp;
+		}
+
+		super(w, h, false, 0);
+		readable = false;
+		image = null;
+
+		var gl = GL.context;
+		var texture = atlasBitmap.getTexture(gl);
+		__parentBitmap = atlasBitmap;
+		__texX0 = x / atlasBitmap.width;
+		__texX1 = (x + width) / atlasBitmap.width;
+		__texY0 = y / atlasBitmap.height;
+		__texY1 = (y + height) / atlasBitmap.height;
+		__texX = x;
+		__texY = y;
+		__texWidth = width;
+		__texHeight = height;
+		__texture = texture;
+		__textureContext = gl;
+		__offsetX = offsetX;
+		__offsetY = offsetY;
+		__rotated = rotated;
+		#else
+		super(0,0,false,0);
+		#end
+	}
+
+	override function clone():BitmapData {
+		return new SubBitmapData(__parentBitmap, __texX, __texY, __texWidth, __texHeight, -__offsetX, -__offsetY, width, height, __rotated);
+	}
+
+	#if !display
+	override function getTexture (gl:GLRenderContext):GLTexture {
+		return __texture;
+	}
+
+	override function getBuffer (gl:GLRenderContext, alpha:Float, colorTransform:ColorTransform):GLBuffer {
+		var __bufferStride = BitmapData.__bufferStride;
+
+		if (__buffer == null || __bufferContext != gl) {
+			__bufferData = new Float32Array (__bufferStride * 4);
+
+			if (__rotated) {
+				__bufferData[0] = __offsetX + __texHeight;
+				__bufferData[1] = __offsetY + __texWidth;
+				__bufferData[2] = 0;
+
+				__bufferData[3] = __texX0;
+				__bufferData[4] = __texY1;
+
+				// ------
+
+				__bufferData[__bufferStride] = __offsetX;
+				__bufferData[__bufferStride + 1] = __offsetY + __texWidth;
+				__bufferData[__bufferStride + 2] = 0;
+
+				__bufferData[__bufferStride + 3] = __texX0;
+				__bufferData[__bufferStride + 4] = __texY0;
+
+				// ------
+
+				__bufferData[__bufferStride * 2] = __offsetX + __texHeight;
+				__bufferData[__bufferStride * 2 + 1] = __offsetY;
+				__bufferData[__bufferStride * 2 + 2] = 0;
+
+				__bufferData[__bufferStride * 2 + 3] = __texX1;
+				__bufferData[__bufferStride * 2 + 4] = __texY1;
+
+				// ------
+
+				__bufferData[__bufferStride * 3] = __offsetX;
+				__bufferData[__bufferStride * 3 + 1] = __offsetY;
+				__bufferData[__bufferStride * 3 + 2] = 0;
+
+				__bufferData[__bufferStride * 3 + 3] = __texX1;
+				__bufferData[__bufferStride * 3 + 4] = __texY0;
+			} else {
+				__bufferData[0] = __offsetX + __texWidth;
```

Comment:
isn't this a `frameWidth`?

Reply:
No, we don't use frame dimensions for rendering at all - just offsets.